src/core/core/rpc_engine.cpp

```diff
@@ -174,15 +174,33 @@ namespace dsn {
     void rpc_client_matcher::on_rpc_timeout(uint64_t key)
     {
         rpc_response_task* call;
-        int bucket_index = key % MATCHER_BUCKET_NR;
+        int bucket_index = key % MATCHER_BUCKET_NR;        
```

Comment:
Your suggestion is usually right. But in this case, it is on a critical path and we deliberately do that (for call and timeout_ts_ms). They will be initialized later and the other case will simply return.